ieeeuoft/hackathon-template
# IEEE-9: Add a default "Application Reviewers" group

## Overview

- Resolves #243 
- [IEEE-9](https://ieeeuoft.atlassian.net/browse/IEEE-9?atlOrigin=eyJpIjoiNjI2YmY0MjAwZDM5NDgwZDhmNWIwZWI1MDBjYjcxOWMiLCJwIjoiaiJ9)
- Added a revertible migration to create an Application Reviewers group
- The group has the following permissions:
   - User view
   - Application view
   - Review add
   - Review change
   - Review view
   - Review Team view
- Modified existing reviewer admin unit tests to use the group, instead of assigning change permissions. "View only" unit tests are unchanged.
    - I realize that these unit tests now check something slightly different from the old ones. In my opinion this is okay and good because going forward the reviewers will not be assigned individual permissions, only the group. Therefore, we need to test the forms with the group. While we could duplicate the tests, once for the group and once for the actual change permissions, that seems like overkill to me.
    - The "view only" tests are used to ensure that users without add/change permission aren't able to load the review forms and have a security function, so I kept them as is.


## Unit Tests Created

- Created a unit test to check if the Application Reviewers group has exactly the permissions specified in `review` app root


## Steps to QA

- Run `python manage.py migrate review` to apply. Login to the website and verify that the group has been created. Assign it to a new user.
    - Make sure the user can view applications, reviews, team reviews, and users. 
    - Make sure the user can add and change, but not delete reviews
    - Make sure the user has no other access
- Run `python manage.py migrate review 0002` to revert the migration. Login and make sure the group has been deleted.
